src/_pytest/config/__init__.py

```diff
@@ -378,25 +378,27 @@ def _try_load_conftest(self, anchor):
     def _getconftestmodules(self, path):
         if self._noconftest:
             return []
+
+        if path.isfile():
```

Comment:
just use if/else to set directory instead of recursing

Reply:
Yeah.
Do you agree with the change in general though? (to only cache dirs)

Reply:
Cool, amended.